File: C_Programming/Practise_Programs/PP_109.c
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_109
//
//  Description       : This program demonstrates string input using
//                      scanf() with a scanset to accept spaces in
//                      the input string.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 13/08/2026
//
//  Time Complexity   : O(n)
//  Space Complexity  : O(n)
//
//////////////////////////////////////////////////////////////////

#include<stdio.h>

int main()
{
    char Arr[50] = {'\0'};

    printf("Enter string : \n");
    scanf("%[^'\n']s", Arr);    // Regular expression (Regex)       

    printf("Enter string is : %s\n", Arr);
    
    return 0;
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input :
//      Jay Ganesh
//
//  Sample Output :
//
//      Enter string :
//      Enter string is : Jay Ganesh
//
//////////////////////////////////////////////////////////////////

File: Java_Programming/Practise_Programs/PP_109.java
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_109
//
//  Description       : This program accepts a string from the user
//                      and displays its first three characters using
//                      the charAt() method.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 26/08/2026
//
//  Time Complexity   : O(1)
//  Space Complexity  : O(n)
//
//////////////////////////////////////////////////////////////////

import java.util.Scanner;

class PP_109
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);
        String Arr = null;

        System.out.println("Enter String : ");
        Arr = sobj.nextLine();

        System.out.println(Arr.charAt(0));
        System.out.println(Arr.charAt(1));
        System.out.println(Arr.charAt(2));

        sobj.close();
    }
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input :
//      Ganesh
//
//  Sample Output :
//
//      Enter String :
//      G
//      a
//      n
//
//////////////////////////////////////////////////////////////////